nixos/modules/security/pass.nix

```diff
@@ -0,0 +1,200 @@
+          RFC2136_TSIG_KEY=acme_example_org
+          RFC2136_PROPAGATION_TIMEOUT=1000
+          RFC2136_POLLING_INTERVAL=30
+          RFC2136_SEQUENCE_INTERVAL=30
+          RFC2136_DNS_TIMEOUT=1000
+          RFC2136_TTL=1
+          ''}
+        ";
+      };
+    };
+    type = types.attrsOf (types.submodule ({name, config, ...}: {
+      options = {
+        gpg = lib.mkOption {
+          type = types.path;
+          default = builtins.path {
+            path = pass.store + "/${name}.gpg";
+            name = "${escapeUnitName name}.gpg";
+          };
+          description = ''
+            The path to the gnupg-encrypted secret.
+            It will be copied into the Nix store of the orchestrating and of the target system.
+            It must be decipherable by an OpenPGP key within <literal>gnupgHome</literal>,
+            whose passhrase is on the target system into <literal>passphraseFile</literal>.
+            Defaults to the name of the secret, prefixed by <literal>passwordStore</literal>
+            and suffixed by <literal>.gpg</literal>.
+          '';
+        };
+        gnupgHome = lib.mkOption {
+          type = types.str;
+          default = "/root/.gnupg";
+          description = ''
+            The directory on the target system to the <literal>gnupg</literal> home
+            used to decrypt the secret.
+          '';
+        };
+        passphraseFile = lib.mkOption {
+          type = types.str;
+          default = "/root/key.pass";
+          description = ''
+            The directory on the target system to a file containing
+            the password of an OpenPGP key in <literal>gnupgHome</literal>,
+            to which <literal>gpg</literal> secret is encrypted to.
+          '';
+        };
+        mode = lib.mkOption {
+          type = types.str;
+          default = "400";
+          description = ''
+            Permission mode of the secret <literal>path</literal>.
+          '';
+        };
+        user = lib.mkOption {
+          type = types.str;
+          default = "root";
+          description = ''
+            Owner of the secret <literal>path</literal>.
+          '';
+        };
+        group = lib.mkOption {
+          type = types.str;
+          default = "root";
+          description = ''
+            Group of the secret <literal>path</literal>.
+          '';
+        };
+        pipe = lib.mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          description = ''
+            Shell command taking the deciphered secret on its standard input
+            and which must put on its standard output
+            the actual material to be installed.
+            This allows to decorate the secret with non-secret bits.
+          '';
+        };
+        path = lib.mkOption {
+          type = types.str;
+          default = name;
+          apply = p: if match "^/.*" p == null then "/run/pass-secrets/"+p+"/file" else p;
```

Comment:
Maybe put this on ramfs.

Reply:
Isn't it already the case? on my NixOS I have:
```
tmpfs on /run type tmpfs (rw,nosuid,nodev,size=1990820k,mode=755)
```

Reply:
oh, right, `/run/keys` really is `ramfs`, whereas `tmpfs` may put files on the swap. I'll fix that tomorrow. Thanks!